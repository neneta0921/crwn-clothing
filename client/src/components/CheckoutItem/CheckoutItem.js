import { connect } from 'react-redux';

import { clearItemFromCart, addItem, removeItem } from '../../redux/cart/cartActions';

import {
  CheckoutItemContainer,
  ImageContainer,
  QuantityContainer,
  RemoveButtonContainer,
  TextContainer,
} from './CheckoutItemStyles';

export const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
  const { name, imageUrl, price, quantity } = cartItem;

  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item" />
      </ImageContainer>
      <TextContainer>{name}</TextContainer>
      <QuantityContainer>
        <div onClick={() => removeItem(cartItem)}>&#45;</div>
        <span>{quantity}</span>
        <div onClick={() => addItem(cartItem)}>&#43;</div>
      </QuantityContainer>
      <TextContainer>{price}</TextContainer>
      <RemoveButtonContainer onClick={() => clearItem(cartItem)}>&#10005;</RemoveButtonContainer>
    </CheckoutItemContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  clearItem: (item) => dispatch(clearItemFromCart(item)),
  addItem: (item) => dispatch(addItem(item)),
  removeItem: (item) => dispatch(removeItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
