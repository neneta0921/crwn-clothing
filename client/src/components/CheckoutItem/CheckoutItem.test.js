import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { CheckoutItem } from './CheckoutItem';

describe('CheckoutItem component', () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();
    const mockProps = {
      cartItem: {
        imageUrl: 'www.testImage.com',
        price: 20,
        name: 'hats',
        quantity: 3,
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
    };
    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it('should render CheckoutItem component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call clearItem when remove button is clicked', () => {
    wrapper.find('.remove-button').simulate('click');
    expect(mockClearItem).toHaveBeenCalled();
  });

  it('should call removeItem when left minus is clicked', () => {
    wrapper.find('.quantity-container').childAt(0).simulate('click');
    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it('should call addItem when rihgt plus is clicked', () => {
    wrapper.find('.quantity-container').childAt(2).simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });
});
