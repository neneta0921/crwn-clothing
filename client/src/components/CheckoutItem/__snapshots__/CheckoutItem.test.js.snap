// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should render CheckoutItem component 1`] = `
<div
  className="sc-bdfBwQ eUoWEP"
>
  <div
    className="sc-gsTCUz fAhmBG"
  >
    <img
      alt="item"
      src="www.testImage.com"
    />
  </div>
  <span
    className="sc-dlfnbm dqsrEz"
  >
    hats
  </span>
  <span
    className="sc-dlfnbm sc-hKgILt dqsrEz lhoYWa"
  >
    <div
      onClick={[Function]}
    >
      -
    </div>
    <span>
      3
    </span>
    <div
      onClick={[Function]}
    >
      +
    </div>
  </span>
  <span
    className="sc-dlfnbm dqsrEz"
  >
    20
  </span>
  <div
    className="sc-eCssSg dJkOIt"
    onClick={[Function]}
  >
    âœ•
  </div>
</div>
`;
